GitPrompt.ps1

```diff
@@ -82,6 +82,9 @@ $global:GitPromptSettings = New-Object PSObject -Property @{
 
     AutoRefreshIndex                            = $true
 
+    # Valid values are "Full", "Compact", and "Minimal"

```

Comment:
That change that was proposed has a side effect of having to pass the same parameters that you've already changed to the config method, since the default value for the parameter is a hard-coded defined value.  You could still save the settings in a variable/property bag for storing, and perhaps have the default value for the parameter be the value already in the property bag, so that once changed, it will be used for subsequent sets.

Or you can do the Add-Type -TypeDefinition method to create enums you need and a class that has a bunch of public properties.  They won't have tab-completion, but they will have parameter validation.